pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -847,18 +849,62 @@ func TestGetProxyServiceInstances_WorkloadInstance(t *testing.T) {
 	cache.WaitForCacheSync(ctl.stop, ctl.HasSynced)
 	defer ctl.Stop()
 
+	createService(ctl, "ratings", "bookinfo-ratings",
+		map[string]string{
+			annotation.AlphaKubernetesServiceAccounts.Name: "ratings",
+			annotation.AlphaCanonicalServiceAccounts.Name:  "ratings@gserviceaccount2.com",
+		},
+		[]int32{8080}, map[string]string{"app": "ratings"}, t)
+
+	ev := fx.Wait("service")
+	if ev == nil {
+		t.Fatal("Timeout creating service")
+	}
+
+	createService(ctl, "details", "bookinfo-details",
+		map[string]string{
+			annotation.AlphaKubernetesServiceAccounts.Name: "details",
+			annotation.AlphaCanonicalServiceAccounts.Name:  "details@gserviceaccount2.com",
+		},
+		[]int32{9090}, map[string]string{"app": "details"}, t)
+
+	ev = fx.Wait("service")
+	if ev == nil {
+		t.Fatal("Timeout creating service")
+	}
+
 	createService(ctl, "reviews", "bookinfo-reviews",
 		map[string]string{
 			annotation.AlphaKubernetesServiceAccounts.Name: "reviews",
 			annotation.AlphaCanonicalServiceAccounts.Name:  "reviews@gserviceaccount2.com",
 		},
 		[]int32{7070}, map[string]string{"app": "reviews"}, t)
 
-	ev := fx.Wait("service")
+	ev = fx.Wait("service")
```

Comment:
very nit - may be easier to just pass in `t` to Wait if we are always just going to fail the same way

Reply:
done